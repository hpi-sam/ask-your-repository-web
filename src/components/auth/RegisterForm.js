// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Form from 'react-validation/build/form';
import Input from 'react-validation/build/input';
import Button from 'react-validation/build/button';
import type { UserParams } from '../../models/User';
import { register } from '../../state/auth/auth.actionCreators';
import './RegisterForm.scss';

type Props = {
    dispatch: Function,
}

type State = {
    user: UserParams,
}

class RegisterForm extends Component<Props, State> {
  constructor(props) {
    super(props);

    this.state = {
      user: {
        email: '',
        username: '',
        password: '',
      },
    };
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    const { user } = this.state;
    this.setState({
      user: {
        ...user,
        [name]: value,
      },
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();

    const { user } = this.state;
    const { dispatch } = this.props;
    if (user.username && user.email && user.password) {
      dispatch(register(user));
    }
  }

  render() {
    const { user } = this.state;
    return (
      <Form onSubmit={this.handleSubmit} className="RegisterForm">
        <h1>Register</h1>
        <div className="form-input">
          <label>
            Username:
            <Input
              type="text"
              className="form-control"
              name="username"
              value={user.username}
              onChange={this.handleChange}
              data-cy="register-username-input"
            />
          </label>
        </div>
        <div className="form-input">
          <label>
            Email:
            <Input
              type="text"
              className="form-control"
              name="email"
              value={user.email}
              onChange={this.handleChange}
              data-cy="register-email-input"
            />
          </label>
        </div>
        <div className="form-input">
          <label>
            Password:
            <Input
              type="password"
              className="form-control"
              name="password"
              value={user.password}
              onChange={this.handleChange}
              data-cy="register-password-input"
            />
          </label>
        </div>
        <div className="form-input">
          <label>
            Repeat Password:
            <Input
              type="password"
              className="form-control"
              name="passwordRepeat"
              onChange={this.handleChange}
              data-cy="register-password-repeat-input"
            />
          </label>
        </div>

        <div className="RegisterForm__buttons">
          <Button data-cy="register-submit-button">
            Register
          </Button>
          <Link to="/login" className="cancel">Cancel</Link>
        </div>
      </Form>
    );
  }
}

export default connect()(RegisterForm);
