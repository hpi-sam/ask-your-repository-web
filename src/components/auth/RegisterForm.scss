.RegisterForm {
  margin: 40px auto;
  width: 80%;
  max-width: 540px;
  padding: 0 5%;

  h1 {
    font-weight: bold;
  }

  &__buttons {
    display: flex;

    button {
      display: inline-block;
      background-color: green;
      margin-top: 1em;
      width: 48%;
      font-weight: bold;
      padding: 1rem 2rem;
      border-radius: .3rem;
    }

    .cancel {
      background-color: gray;
      display: inline-block;
      color: white;
      margin-top: 1em;
      margin-left: 4%;
      width: 48%;
      font-weight: bold;
      padding: 1rem 2rem;
      border-radius: .3rem;
      text-align: center;
    }
  }
}
