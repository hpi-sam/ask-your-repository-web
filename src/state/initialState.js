// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as presentationMode } from './presentation_mode/presentationMode.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as auth } from './auth/auth.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  activeTeam,
  teamSidebar,
  presentationMode,
  auth,
};

export default initialState;
