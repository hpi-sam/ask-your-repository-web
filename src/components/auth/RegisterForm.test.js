// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import RegisterForm from './RegisterForm';
import initialState from '../../state/initialState';

const mockStore = configureMockStore();

describe('<RegisterForm />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount((
      <Provider store={store}>
        <MemoryRouter>
          <RegisterForm />
        </MemoryRouter>
      </Provider>
    )).find('RegisterForm');
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
