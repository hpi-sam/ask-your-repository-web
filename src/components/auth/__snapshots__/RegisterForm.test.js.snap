// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm /> renders correctly 1`] = `
<RegisterForm
  dispatch={[Function]}
>
  <Form(t)
    className="RegisterForm"
    onSubmit={[Function]}
  >
    <t
      className="RegisterForm"
      getValues={[Function]}
      hideError={[Function]}
      onSubmit={[Function]}
      showError={[Function]}
      validate={[Function]}
      validateAll={[Function]}
    >
      <form
        className="RegisterForm"
        onSubmit={[Function]}
      >
        <h1>
          Register
        </h1>
        <div
          className="form-input"
        >
          <label>
            Username:
            <Control(s)
              className="form-control"
              data-cy="register-username-input"
              name="username"
              onChange={[Function]}
              type="text"
              validations={Array []}
              value=""
            >
              <s
                className="form-control"
                data-cy="register-username-input"
                name="username"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              >
                <div>
                  <input
                    className="form-control"
                    data-cy="register-username-input"
                    name="username"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div
          className="form-input"
        >
          <label>
            Email:
            <Control(s)
              className="form-control"
              data-cy="register-email-input"
              name="email"
              onChange={[Function]}
              type="text"
              validations={Array []}
              value=""
            >
              <s
                className="form-control"
                data-cy="register-email-input"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              >
                <div>
                  <input
                    className="form-control"
                    data-cy="register-email-input"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div
          className="form-input"
        >
          <label>
            Password:
            <Control(s)
              className="form-control"
              data-cy="register-password-input"
              name="password"
              onChange={[Function]}
              type="password"
              validations={Array []}
              value=""
            >
              <s
                className="form-control"
                data-cy="register-password-input"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                type="password"
                value=""
              >
                <div>
                  <input
                    className="form-control"
                    data-cy="register-password-input"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div
          className="form-input"
        >
          <label>
            Repeat Password:
            <Control(s)
              className="form-control"
              data-cy="register-password-repeat-input"
              name="passwordRepeat"
              onChange={[Function]}
              type="password"
              validations={Array []}
            >
              <s
                className="form-control"
                data-cy="register-password-repeat-input"
                name="passwordRepeat"
                onBlur={[Function]}
                onChange={[Function]}
                type="password"
                value=""
              >
                <div>
                  <input
                    className="form-control"
                    data-cy="register-password-repeat-input"
                    name="passwordRepeat"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </s>
            </Control(s)>
          </label>
        </div>
        <div
          className="RegisterForm__buttons"
        >
          <Button(s)
            data-cy="register-submit-button"
          >
            <s
              data-cy="register-submit-button"
              hasErrors={false}
            >
              <button
                data-cy="register-submit-button"
                disabled={false}
              >
                Register
              </button>
            </s>
          </Button(s)>
          <Link
            className="cancel"
            replace={false}
            to="/login"
          >
            <a
              className="cancel"
              href="/login"
              onClick={[Function]}
            >
              Cancel
            </a>
          </Link>
        </div>
      </form>
    </t>
  </Form(t)>
</RegisterForm>
`;
